Model/ModelManager.php

```diff
@@ -385,8 +386,12 @@ public function getNormalizedIdentifier($entity)
             throw new \RunTimeException('Invalid argument, object or null required');
         }
 
-        // the entities is not managed
-        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
+        if (!$entity) {
+            return;
+        }
+
+        $entityState = $this->getEntityManager($entity)->getUnitOfWork()->getEntityState($entity);
+        if (UnitOfWork::STATE_NEW == $entityState || UnitOfWork::STATE_REMOVED == $entityState) {
```

Comment:
I have no problem with this but, in my defence:
 - `getEntityState()` says that is will only return integers, and
 - would we really care if it returns '2' as a string for some reason instead of 2 as an int? (or '4')?

Secondly, I've always been told to avoid in_array unless it's really necessary (as used in #633 ). I think it's for performance reasons, but maybe I've just been doing this too long and it's not an issue any more!